Client/Frontend/UserContent/UserScripts/AllFrames/AtDocumentStart/AdsReporting.js

```diff
@@ -0,0 +1,26 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use strict";
+
+if (webkit.messageHandlers.adsMediaReporting) {
+    install();
+}
+
+function install() {
+  function sendMessage(playing) {
+    webkit.messageHandlers.adsMediaReporting.postMessage(playing);
+  }
+
+  let originalPlay = HTMLMediaElement.prototype.play;
+  HTMLMediaElement.prototype.play = function() {
+    this.addEventListener('playing', function() {
```

Comment:
It is very important to removeEventListener before adding another one. This will lead to multiple listeners while the user browses. To test open YouTube and open a video. after video plays tap another video from suggestions. You will find more breakpoint hits(Equal to the number of new videos played) 
FYI since the property we hooked is js play you can't reproduce it from the play button in iPhone(Its a Video player button) but can easily reproduce it with play button on iPad with Desktop UA

Reply:
Another issue I found is that this does not work if the video is not played from js but rather requires user action. To test, host this and open in brave:

`<!DOCTYPE html>
<html>
<body>

<button onclick="playVid()" type="button">Play Video</button>
<button onclick="pauseVid()" type="button">Pause Video</button><br> 

<video id="myVideo" width="320" height="176" controls playsinline autoplay>
  <source src="https://www.w3schools.com/html/mov_bbb.mp4" type="video/mp4">
  <source src="mov_bbb.ogg" type="video/ogg">
  Your browser does not support HTML5 video.
</video>

<script> 
var vid = document.getElementById("myVideo"); 

function playVid() { 
  vid.play(); 
} 

function pauseVid() { 
  vid.pause(); 
} 
</script> 

<p>Video courtesy of <a href="https://www.bigbuckbunny.org/" target="_blank">Big Buck Bunny</a>.</p>
</body>
</html>

`

Reply:
Another Issue: For some reason the Webview is not purged when you close the tab. To repro:
-> Open Dailymotion and play a video
-> Once you can hear the audio, close the tab
-> Even when the tab is closed you can hear the audio.

Reply:
vid.play() works, tapping on video player play button does not(If the video has not been played by js ie `play()` yet)

Reply:
they have tap event listeners, this is why I pasted the html code above for testing. When you tap the native play button it does not call the js Play() function rather it does it internally.